src/v/kafka/server/group.cc

```diff
@@ -2918,7 +2918,45 @@ ss::future<> group::do_try_abort_old_tx(model::producer_identity pid) {
         }
     }
 
-    // Add logic to recommit/abort txs
+    auto p_it = _prepared_txs.find(pid);
+    if (p_it != _prepared_txs.end()) {
+        auto r = co_await _tx_frontend.local().try_abort(
+          model::partition_id(
+            0), // TODO: Pass partition_id to prepare request and use it here.
+                // https://github.com/redpanda-data/redpanda/issues/6137
+          pid,
+          p_it->second.tx_seq,
+          config::shard_local_cfg().rm_sync_timeout_ms.value());
+        if (r.ec == cluster::tx_errc::none) {
+            if (r.commited) {
+                auto res = co_await do_commit(_id, pid);
+                if (res.ec != cluster::tx_errc::none) {
+                    vlog(
+                      _ctxlog.error,
+                      "Can not commit prepared tx with pid({})",
+                      pid);
+                }
+            } else if (r.aborted) {
+                auto res = co_await do_abort(_id, pid, p_it->second.tx_seq);
```

Comment:
```
auto res = co_await do_abort(_id, pid, p_it->second.tx_seq);
```

in this statement, what guarantees that `p_it->second` is still valid to access after the scheduling point introduced above with `co_await try_abort`? it looks like this could be a potential segfault if the _prepared_txs hash table were to be rehashed or undergo some other change that lead to the iterator being invalidated.

it looks like a per-pid lock is held, but that data structure exists across pids.

Reply:
Oh
Thanks!!!